### PR TITLE
Move compiler compatibility checks into CMake

### PR DESCRIPTION
**Pull request by BenjaminW3**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,24 +55,24 @@ Accelerator Back-ends
 | std::thread | std::thread |Host CPU (multi core)|sequential|parallel (preemptive multitasking)|
 | Boost.Fiber | boost::fibers::fiber |Host CPU (single core)|sequential|parallel (cooperative multitasking)|
 |TBB 2.2+ blocks|TBB 2.2+|Host CPU (multi core)|parallel (preemptive multitasking)|sequential (only 1 thread per block)|
-|CUDA 8.0+|CUDA 8.0+|NVIDIA GPUs SM 2.0+|parallel (undefined)|parallel (lock-step within warps)|
+|CUDA 8.0-9.1|CUDA 8.0-9.1|NVIDIA GPUs SM 2.0+|parallel (undefined)|parallel (lock-step within warps)|
```
Comment (ax3l) [inline README.md]: note: SM 3.0+ for CUDA 9.0+

File: alpakaConfig.cmake [hunk 2]
```diff
@@ -364,6 +378,40 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
...
 
             ELSE()
+                IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+                    IF(CUDA_VERSION VERSION_EQUAL 8.0)
+                        IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 5.0)
+                            MESSAGE(FATAL_ERROR "NVCC 8.0 does not support GCC 6+ and fails compiling the std::tuple implementation in GCC 5+. Please use GCC 4.9!")
+                        ENDIF()
+                    ELSEIF((CUDA_VERSION VERSION_EQUAL 9.0) OR (CUDA_VERSION VERSION_EQUAL 9.1))
+                        IF(CMAKE_CXX_COMPILER_VERSION GREATER_EQUAL 6.0)
+                            MESSAGE(FATAL_ERROR "NVCC 9.0-9.1 do not support GCC 7+ and fail compiling the std::tuple implementation in GCC 6+. Please use GCC 4.9 or 5!")
```
Comment (ax3l) [inline alpakaConfig.cmake]: fails

**Comment by BenjaminW3** [comment]
@ax3l Should we port those changes back to a possible 0.3.2? We would have to resolve many merge conflicts because the 0.3.* branch does still support CUDA 7.

**Comment by ax3l** [comment]
@BenjaminW3 yes, let us at least collect them in a potential 0.3.2 release project

**Comment by BenjaminW3** [comment]
Ready to be merged!

**Pull request merged**